src/app/questionnaire-item/store/transform-response.ts

```diff
@@ -154,41 +164,48 @@ const getResponseItems: (
 ) => (items: QuestionnaireItem[]) => fhir.r4.QuestionnaireResponseItem[] = (
   getIsItemEnabled
 ) =>
-  R.chain((item) => {
-    const isEnabled = getIsItemEnabled(item);
-    const getIsEnabledForSubItems = isEnabled
-      ? getIsItemEnabled
-      : R.always(false);
-    const responseItem: fhir.r4.QuestionnaireResponseItem = {
-      linkId: item.linkId,
-      text: item.text,
-    };
-    const getItems = R.pipe(
-      R.values,
-      getResponseItems(getIsEnabledForSubItems)
-    );
-    if (item.type === 'group') {
-      return R.map(
-        ({ items }) => ({
+  R.pipe<
+    QuestionnaireItem[],
+    QuestionnaireItem[],
+    fhir.r4.QuestionnaireResponseItem[],
+    fhir.r4.QuestionnaireResponseItem[]
+  >(
+    R.filter(getIsItemEnabled),
+    R.chain((item) => {
+      const responseItem: fhir.r4.QuestionnaireResponseItem = {
+        linkId: item.linkId,
+        text: item.text,
+      };
+      const getItems = R.pipe(R.values, getResponseItems(getIsItemEnabled));
+      if (item.type === 'group') {
+        return R.pipe<
+          readonly ItemAnswer[],
+          fhir.r4.QuestionnaireResponseItem[],
+          fhir.r4.QuestionnaireResponseItem[]
+        >(
+          R.map(({ items }) => ({
+            ...responseItem,
+            item: getItems(items),
+          })),
+          R.filter<fhir.r4.QuestionnaireResponseItem>(
+            ({ item: answerItem }) =>
+              !R.isNil(answerItem) && !R.isEmpty(answerItem)
+          )
+        )(item.itemAnswerList);
+      }
+      const responseAnswer = getResponseAnswers(item, getIsItemEnabled);
+      if (R.isEmpty(responseAnswer)) {
+        return;
+      }
+      return [
```

Comment:
```suggestion
      return R.isEmpty(responseAnswer) ? [] : [
```
If you change it like this you can remove the filter call below.